File: MyRixi-API/Models/Comment.cs
namespace MyRixiApi.Models;

public class Comment
{
    public Guid Id { get; set; }
    public string Content { get; set; } = string.Empty;
    public DateTime PostedAt { get; set; } = DateTime.UtcNow;
    public DateTime? EditedAt { get; set; }
    public bool IsEdited { get; set; } = false;
    
    public Guid SenderId { get; set; }
    public MainProfile? Sender { get; set; }
    
    public Guid? ParentCommentId { get; set; }
    public Comment? ParentComment { get; set; }
    public Guid? PostId { get; set; }
    public Post? Post { get; set; }
    public Guid? ProfileId { get; set; }
    public MainProfile? Profile { get; set; }
    public ICollection<Comment> Replies { get; set; } = new List<Comment>();
}

File: MyRixi-Mobile/types/comment.ts
export interface Comment {
  id: string;
  content: string;
  postedAt: string;
  parentCommentId?: string;
  postId?: string;
  profileId: string;
  profileDisplayName: string;
  profilePictureUrl: string;
  repliesCount: number;
  isOwner: boolean;
}